__tests__/integration/helpers/wallet.helper.js

```diff
@@ -50,10 +50,32 @@ const startedWallets = [];
 
 /**
  * Generates a Wallet from an available precalculated seed
+ *
+ * @param [param] Optional object with properties to override the generated wallet
+ * @param {string} [param.seed] 24 words separated by space
+ * @param {string} [param.passphrase=''] Wallet passphrase
+ * @param {string} [param.xpriv]
+ * @param {string} [param.xpub]
+ * @param {string} [param.tokenUid] UID of the token to handle on this wallet
+ * @param {string} [param.password] Password to encrypt the seed
+ * @param {string} [param.pinCode] PIN to execute wallet actions
+ * @param {boolean} [param.debug] Activates debug mode
+ * @param {{pubkeys:string[],numSignatures:number}} [param.multisig]
+ * @param {string[]} [param.preCalculatedAddresses] An array of pre-calculated addresses
+ *
  * @returns {Promise<HathorWallet>}
  */
-export async function generateWalletHelper() {
-  const walletData = precalculationHelpers.test.getPrecalculatedWallet();
+export async function generateWalletHelper(param) {
+  /** @type PrecalculatedWalletData */
+  let walletData = {};
+
+  // Only fetch a precalculated wallet if the input does not offer a specific one
+  if (!param) {
```

Comment:
That way if I didn't want to pass a `preCalculatedAddresses` parameters I would have to explicitly set it to `[]` on the parameters, otherwise the helper would consume another wallet and yield non-obvious results. A test using a `xpub` and `preCalculatedAddresses` without a seed would not at all be possible.

The rational for the current implementation is that:
- If the helper method is called without parameters, it will generate a valid wallet
- If the helper is called with a parameter, the user will have to define the exact configuration for the wallet
- The rule is as easy as possible for a developer to just skim the code and understand it

After all, initializing a wallet without the helper isn't complex. What do you think?

Reply:
Now I understand, I got confused by the method docstring

"Generates a Wallet from an available precalculated seed"

I thought you wanted only pre calculated wallets. I just think you should update the docstring.

Reply:
Updated on [2305663](https://github.com/HathorNetwork/hathor-wallet-lib/pull/409/commits/23056630d7b67c9881c1194ce588d6c21cf3d9dc).